File: b_rank_skillcheck_archive/kimariji_04.py
# 実行時間 10ms
o = 0
n, m = list(map(int, input().split()))
s = []

for _ in [0] * n:
    s.append(input())
for i in range(min(len(s[0]), len(s[1]))):
    o += 1
    if s[0][i] != s[1][i]:
        break
print(s[m - 1][:o])

File: b_rank_skillcheck_archive/kimariji_04.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int o = 0;
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();
        string[] s = new string[n[0]];

        foreach( int i in Range( 0, n[0] ) ) {
            s[i] = ReadLine();
        }
        foreach( int i in Range( 0, System.Math.Min( s[0].Length, s[1].Length ) ) ) {
            o++;
            if( s[0][i] != s[1][i] ) {
                break;
            }
        }
        WriteLine( s[n[1] - 1].Substring( 0, o ) );
        return;
    }
}
